Sources/SwiftBuild/SWBuildMessage+Protocol.swift

```diff
@@ -48,7 +48,9 @@ extension SwiftBuildMessage {
     init(_ message: BuildOperationEnded) {
         let metrics: BuildOperationMetrics?
         if let messageMetrics = message.metrics, !messageMetrics.counters.isEmpty {
```

Comment:
This check will omit the metrics if `counters` is empty even if `taskCounters` is not. Maybe add a `BuildOperationMetrics.isEmpty` function that will check for both?

Reply:
Awesome, thanks for catching that! I [updated it](https://github.com/swiftlang/swift-build/compare/d544a5abf80de99f4214c26aa3b6f7c904d79d93..c8b3c74349eb2bff65b693bd9c657488c3f31d9a) but not as suggested, I didn't want nil to have ambiguous meaning (was it empty or nil for another reason), but I do propagate nil input (which is used for build cancellation and some types of build failures). Does that work?

Reply:
Sounds good!